### ISSUE TITLE
genus and higher level taxon rank detecting

### ISSUE THREAD

**Issue by fontikar** [traitecoevo/APCalign#101]
```
create_taxonomic_update_lookup( 
+   taxa = c(
+     "Banksia integrifolia",
+     "Acacia longifolia",
+     "Commersonia rosea",
+     "Acacia sp."
+     ),
+   resources = resources
+ )
Checking alignments of 4 taxa

  -> 0 names already matched; 0 names checked but without a match; 4 taxa yet to be checked
# A tibble: 4 × 7
  original_name        aligned_name         accepted_name taxon_rank author aligned_reason updated_reason
  <chr>                <chr>                <chr>         <chr>      <chr>  <chr>          <chr>         
1 Banksia integrifolia Banksia integrifolia Banksia inte… Species    L.f.   match_06. Aut… accepted      
2 Acacia longifolia    Acacia longifolia    Acacia longi… Species    (Andr… match_06. Aut… accepted      
3 Commersonia rosea    Commersonia rosea    Androcalva r… Species    (S.A.… match_06. Aut… basionym      
4 Acacia sp.           Acacia sp. [XXXX]    NA            NA         NA     match_01. Rew… known_name_bu…
```

**Comment by wcornwell** [comment]
```
create_taxonomic_update_lookup(c("Acacia sp.","Fabaceae","Caesalpinioideae","Fabales","Rosids","Eudicots"),resources=resources)
Checking alignments of 6 taxa

  -> 0 names already matched; 0 names checked but without a match; 6 taxa yet to be checked
# A tibble: 2 × 7
  original_name aligned_name                  accepted_name taxon_rank author aligned_reason              updated_reason
  <chr>         <chr>                         <lgl>         <lgl>      <lgl>  <chr>                       <chr>         
1 Acacia sp.    Acacia sp. [XXXX]             NA            NA         NA     match_01. Rewording taxon … known_name_bu…
2 Fabaceae      Fabaceae sp. [Fabaceae; XXXX] NA            NA         NA     match_21. Rewording name t… known_name_bu…
```

**Comment by wcornwell** [comment]
```
create_taxonomic_update_lookup(c("Actinotus","Acacia sp.","Fabaceae","Caesalpinioideae","Fabales","Rosids","Eudicots"),resources=resources)
Checking alignments of 7 taxa

  -> 0 names already matched; 0 names checked but without a match; 7 taxa yet to be checked
# A tibble: 3 × 7
  original_name aligned_name                    accepted_name taxon_rank author aligned_reason                                                                                                             updated_reason                 
  <chr>         <chr>                           <lgl>         <lgl>      <lgl>  <chr>                                                                                                                      <chr>                          
1 Acacia sp.    Acacia sp. [XXXX]               NA            NA         NA     match_01. Rewording taxon with ending with `sp.` to indicate a genus-level alignment with `APC accepted` name (2023-08-10) known_name_but_not_apc_accepted
2 Actinotus     Actinotus sp. [Actinotus; XXXX] NA            NA         NA     match_20. Rewording name to be recognised as genus rank, with genus accepted by APC (2023-08-10)                           known_name_but_not_apc_accepted
3 Fabaceae      Fabaceae sp. [Fabaceae; XXXX]   NA            NA         NA     match_21. Rewording name to be recognised as family rank, with family accepted by APC (2023-08-10)                         known_name_but_not_apc_accepted

**Comment by wcornwell** [comment]
that's as far as i got on this problem

**Comment by wcornwell** [comment]
from APC: <img width="519" alt="Screenshot 2023-08-10 at 12 57 48 pm" src="https://github.com/traitecoevo/APCalign/assets/5534161/548470cf-c697-4eda-b351-ebae3a3b8850">
non-trivial problem

**Comment by fontikar** [comment]
Something weird about the object. 

```
resources$APC |> filter(taxonRank == "Genus")
Error in filter(resources$APC, taxonRank == "Genus") : 
  object 'taxonRank' not found
  
  APC[,"taxonRank" == "Genus"]
# A tibble: 109,624 × 0
# ℹ Use `print(n = ...)` to see more rows

APC[stringr::str_detect(APC$taxonRank, "Genus"),] 
  ```
  
  Will think about this a little later! Is the problem Acacia sp. not being in the APC so we can't pull the taxonRank out?

**Comment by wcornwell** [comment]
if you use the native pipe it defaults to the "native" `filter` which is `stats::filter`

**Comment by wcornwell** [comment]
<img width="544" alt="Screenshot 2023-08-10 at 2 09 38 pm" src="https://github.com/traitecoevo/APCalign/assets/5534161/6323bc77-0df4-40a5-a63c-e8e06deb8a95">

**Comment by wcornwell** [comment]
`align_taxa` adds a "sp." but then it has to be gone again to find it in APC:
<img width="1323" alt="Screenshot 2023-08-10 at 2 12 15 pm" src="https://github.com/traitecoevo/APCalign/assets/5534161/6c78ea53-c60e-4172-a65f-825d5855e960">

**Comment by wcornwell** [comment]
> This is too big of a fix before the sprint review

yes, i agree.  tricky one

**Comment by fontikar** [comment]
@ehwenk has a solution replacing taxonomic_resolution to taxon_rank! 🤯

**Comment by ehwenk** [comment]
In `match_taxa.R`, a taxon_rank is assigned each time a name is checked. For names where we declare the taxon_rank (e.g. all the genus sp. names & variants), it is assigned either `genus` or `family`. For names that are exact APC or APNI matches, the taxon_rank is pulled from the taxonomic_resources. I think this was confusing because I had it names `taxonomic_resolution` referencing the resolution to which a name could be assigned, rather than the rank of the name. But it is the information on the taxon_rank.

So the solution is that 
* if `taxon_rank %in% c("genus", "family)` then the first word of the aligned name is used to determine the accepted name.
* for other `taxon_rank` values, the entire aligned name is used to determine the accepted name.

This routine was buried somewhere in how AusTraits works.

**Comment by ehwenk** [comment]
Genus and family level alignments will work for all taxon names once the next big branch is merged in. `subclass` is also propagated through from the APC file, but there aren't any manipulations to ensure it is filled in for taxon names that aren't exact matches - for instance, for APNI names with APC-accepted genera or families. If we want to add this, it wouldn't be hard, as similar code already fills in family names at the end of `update_taxonomy`

**Comment by ehwenk** [comment]
@fontikar This should be working on the main branch - can you confirm? I've just run your example to double check

**Comment by fontikar** [comment]
Seems to be working for genus, but adding Fabaceae here is breaking code

```
create_taxonomic_update_lookup( 
+     taxa = c(
+         "Banksia sp",
+         "Fabaeceae sp",
+         "Dryandra sp"
+     ),
+     resources = resources
+ )
Checking alignments of 3 taxa

  -> 0 names already matched; 0 names checked but without a match; 3 taxa yet to be checked
Error in `dplyr::mutate()`:
ℹ In argument: `taxon_ID = taxon_ID_clean`.
Caused by error:
! object 'taxon_ID_clean' not found
Run `rlang::last_trace()` to see where the error occurred.
```

**Comment by fontikar** [comment]
Genus level alignments are great!! [Screenshot 2023-08-22 at 2 36 17 pm](https://github.com/traitecoevo/APCalign/assets/12161802/3263eb8a-8a43-486b-a0cb-c1bc47b41bb6)

**Comment by ehwenk** [comment]
Two things - one a bug I fixed, but also, you typed in "Fabaeceae sp" not "Fabaceae" and we don't fuzzy align family names
